loader/loader.go

```diff
@@ -309,7 +309,13 @@ func Load(configDetails types.ConfigDetails, options ...func(*Options)) (*types.
 // LoadWithContext reads a ConfigDetails and returns a fully loaded configuration as a compose-go Project
 func LoadWithContext(ctx context.Context, configDetails types.ConfigDetails, options ...func(*Options)) (*types.Project, error) {
 	opts := toOptions(&configDetails, options)
-	dict, err := loadModelWithContext(ctx, &configDetails, opts)
+	sch, err := schema.CreateComposeSchema()
+	if err != nil {
+		return nil, err
+	}
+	fctx := context.WithValue(ctx, consts.ComposeSchema{}, sch)
```

Comment:
Is there any specific reason to initialise the Json Schema parser here and not keep the same logic as previously? 🤔

Reply:
This allows to only compile the schema once (stored in context) vs for each and every compose file we parse. This is indeed not required but a reasonable optimization afaict

Reply:
Alternatively I think it's also valid to move the initialisation to `loadYamlModel` and remove it from the context

Reply:
Make sense, I missed the compilation have to be done each time we parse a file 👌